src/test/resources/lib/configFile/initIDProvider-test.js

```diff
@@ -0,0 +1,159 @@
+const test = require('/lib/xp/testing');
+
+
+
+//////////////////////  Mock workaround (test.mock only works once)
+
+const configFileMocks = {};
+test.mock("/lib/configFile/configFile.js", {
+    getAllIdProviderNames: () => configFileMocks.getAllIdProviderNames(),
+    shouldAutoInit: () => configFileMocks.shouldAutoInit(),
+    getConfigForIdProvider: (idProviderName) => configFileMocks.getConfigForIdProvider(idProviderName)
+});
+const setConfigFileMocks = mocks => {
+    Object.keys(configFileMocks).forEach( key => {
+        delete configFileMocks[key];
+    });
+    Object.keys(mocks).forEach( key => {
+        configFileMocks[key] = mocks[key];
+    })
+};
+
+const beanMocks = {};
+test.mock("/lib/configFile/services/bean.js", {
+    getIdProviders: () => beanMocks.getIdProviders(),
+    createIdProvider: (params) => beanMocks.createIdProvider(params)
+});
+const setBeanMocks = mocks => {
+    Object.keys(beanMocks).forEach( key => {
+        delete beanMocks[key];
+    });
+    Object.keys(mocks).forEach( key => {
+        beanMocks[key] = mocks[key];
+    })
+};
```

Comment:
Cant this be done inside a test method? Then this is not needed. 
Eg. move `test.mock("/lib/configFile/service/bean.js", {...})` inside every test method.

Reply:
(same comment) It should be possible, but it isn't really. If done like that, "natively", each mock only applies once for some reason. So that way, different scenarios can't be tested with different mocks AFAICS (which defies the purpose of mocks), so this workaround was needed.